### DISCUSSION TITLE
_CalcBodyAccelerations() in astronomy.py

### DISCUSSION THREAD

**Discussion by DarkSky7** [cosinekitty/astronomy#312] [General]
In Update(), defined ~around line 10345 of astronomy.py, _CalcBodyAccelerations() is called twice, once before acceleration vectors are computed and once immediately after.  I BARELY understand what's going on in that file, but that apparent 'redundancy' feels weird to me - is there no way to squeeze that out of Update()?

**Comment by cosinekitty** [comment]
The reason I do that is to dramatically decrease errors introduced by using a finite time step &Delta;t to approximate an infinitesimal dt. They are not redundant calls because the first call calculates accelerations at a rough estimate of the next position, while the second call calculates accelerations at a much more accurate position estimate.

It looks like it could be optimized, but the weird thing is that this *is* the optimization, LOL!

If you try to eliminate the double calculation, the first thing you will notice is that the accuracy drops off a cliff. The only way to get the accuracy back is to decrease &Delta;t so much that the whole thing takes far longer than what you started with.

If the performance is really a problem, perhaps the Python version isn't the best fit for your application. It is by far the slowest of all the programming languages Astronomy Engine supports.

**Comment by DarkSky7** [comment]
I wondered if that was what you were doing.  Thanks for the insight.
